test/smartwallet/smartWallet.test.ts

```diff
@@ -76,245 +139,360 @@ describe('SmartWallet', function(){
+                nonce: '0',
+                tokenGas: '0',
+                tokenAmount: '0',
+                tokenContract: fakeToken.address
+            });
+            
+            const typedDeployData = new TypedDeployRequestData(HARDHAT_CHAIN_ID, smartWalletFactory.address, deployRequest);
 
-        it.skip('Should not call transfer on sponsored deployment', async function(){
-            const {smartWallet} = await loadFixture(prepareFixture);
+            const suffixData = getSuffixData(typedDeployData);
 
-            await smartWallet.initialize(owner.address, fakeToken.address, worker.address, 0, 0);
+            const privateKey = Buffer.from(owner.privateKey.substring(2, 66), 'hex');
+            const signature = getLocalEip712DeploySignature(typedDeployData, privateKey);
+            
+            fakeToken.transfer.returns(true);
+            
+            await smartWalletFactory.connect(relayHub).relayedUserSmartWalletCreation(
+                deployRequest.request,
+                suffixData,
+                owner.address,
+                signature
+            );            
 
             expect(fakeToken.transfer).not.to.be.called;
-        });
+        })
+
+        it('Should fail to initialize a SmartWallet twice', async function(){
+            const dataTypesToSign = ['bytes2', 'address', 'address', 'uint256'];
+            const valuesToSign = ['0x1910', owner.address, ZERO_ADDRESS, 0 ];
+
+            const signature = signData(dataTypesToSign, valuesToSign);
 
-        it('Should fail to initialize a contract when it is already initialized', async function(){
-            const {smartWallet} = await loadFixture(prepareFixture);
+            await smartWalletFactory.createUserSmartWallet(
+                owner.address,
+                ZERO_ADDRESS,
+                '0',
+                signature
+            );
+
+            const smartWallet = await getAlreadyDeployedSmartWallet();
 
             await expect(
-                smartWallet.initialize(owner.address, fakeToken.address, worker.address, 10, 400000),
+                smartWallet.initialize(owner.address, fakeToken.address, ZERO_ADDRESS, 10, 400000),
                 'Second initialization not rejected'
             ).to.be.revertedWith('already initialized');
         });
 
         it('Should create the domainSeparator', async function () {
-            const  {smartWallet} = await loadFixture(prepareFixture);
+            const dataTypesToSign = ['bytes2', 'address', 'address', 'uint256'];
+            const valuesToSign = ['0x1910', owner.address, ZERO_ADDRESS, 0 ];
+            
+            const signature = signData(dataTypesToSign, valuesToSign);
+
+            await smartWalletFactory.createUserSmartWallet(
+                owner.address,
+                ZERO_ADDRESS,
+                '0',
+                signature
+            );
+
+            const smartWallet = await getAlreadyDeployedSmartWallet();
```

Comment:
From what I see, these lines are repeated in some tests, so I'm wondering if we can group those tests and include the repeated code in a `beforeEach` function